### PR TITLE
Feature/onramp-agg-integration

### PR DESCRIPTION
**Pull request by alaahd**
**Description of integration tasks**

**TX Details**
- [x] Merge the feature branch of Tx details.
- [x] Conditional rendering for the new Tx details screen based on order type while maintaining backward compatibility.  

**SDK Bootstrapping**
- [x] Remove the provider from the FiatOrders HOC and the Flow in the MainNavigator.
- [x] Keep track of the getStarted view.
- [x] Use the GetStarted screen for bootstrapping the SDK and conditionally rendering the getStarted UI.

**Checklist**
* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Progresses #???

### CODE REVIEW

File: app/components/Views/FiatOrdersView/index.js [hunk 1]
```diff
@@ -26,38 +31,57 @@ const createStyles = (colors) =>
...
+							onPress={() => {
+								item.provider === FIAT_ORDER_PROVIDERS.AGGREGATOR
+									? handleNavigateToTxDetails(item.id)
+									: toggleModal();
+							}}
+							underlayColor={colors.background.alternative}
+							activeOpacity={1}
+						>
+							<OrderListItem order={item} />
+						</TouchableHighlight>
```
Comment (wachunei) [inline app/components/Views/FiatOrdersView/index.js:68]: For cleaner code let's branch this out at a top level, as it is the <Modal> is being mounted for Aggregator Orders.
I'd suggest:
```jsx
	const renderItem = ({ item }) => {
		if(item.provider === FIAT_ORDER_PROVIDERS.AGGREGATOR) {
			return (<TouchableHighlight...)
		}
		// previous code
		return (<ModalHandler...)
	}
```

File: app/components/UI/FiatOnRampAggregator/Views/TransactionDetails.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	dateAndTime: 'FEB 2 2022, 5:05 PM UTC',
+	paymentMethod: 'DEBIT CARD 2755',
+	tokenAmount: '0.0687202',
+	fiatAmount: '$280.54',
+	exchangeRate: '1 ETH @ $4,3427.86',
+	totalFees: '$19.46',
+	providerName: 'MoonPay',
+	purchaseAmountTotal: '$300 USD',
+	paymentType: 'bank',
+};
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/Views/TransactionDetails.js:32]: Remove this

File: app/components/UI/FiatOnRampAggregator/components/Quote.tsx [hunk 3]
```diff
@@ -61,14 +61,16 @@ interface Props {
...
 	};
 	onPress?: () => any;
 	onPressBuy?: () => any;
 	highlighted?: boolean;
+	showInfo: () => any;
 }
 
 const Quote: React.FC<Props> = ({ quote, onPress, onPressBuy, highlighted }: Props) => {
-	const { networkFee, providerFee, amountIn, amountOut, fiat, providerName, crypto } = quote;
+	const { networkFee, providerFee, amountIn, amountOut, fiat, providerName, crypto, showInfo } = quote;
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/components/Quote.tsx:73]: In `Views/GetQuotes` the `showInfo` is passed prop and not part of `quote` object.

File: app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+		},
+		contactDesc: {
+			flexDirection: 'row',
+			alignSelf: 'center',
+			paddingTop: 15,
+		},
+	});
+
+const renderStage = (stage: string, paymentType: string) => {
+	// eslint-disable-next-line react-hooks/rules-of-hooks
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx:66]: Convert this into a Component.

File: app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+					</Text>
+					<ListItem.Content style={styles.listItems}>
+						<ListItem.Body>
+							<Text black small>
+								Token Amount
+							</Text>
+						</ListItem.Body>
+						<ListItem.Amount>
+							<Text small bold primary>
+								{order.data.cryptoAmount} {order.cryptocurrency}
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx:190]: Use render utils, reference: https://github.com/MetaMask/metamask-mobile/pull/4023/commits/cbed8e99a663962897c2d7615872396dc0a991e2

Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx:190]: Use render utils, reference: [`cbed8e9` (#4023)](https://github.com/MetaMask/metamask-mobile/pull/4023/commits/cbed8e99a663962897c2d7615872396dc0a991e2)

File: app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+					<ListItem.Content style={styles.listItems}>
+						<ListItem.Body>
+							<Text black small>
+								{order.currency} Amount
+							</Text>
+						</ListItem.Body>
+						<ListItem.Amount>
+							<Text small bold primary>
+								{order.data.fiatCurrency.denomSymbol}
+								{order.data.fiatAmount}
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx:203]: Use render utils, reference: [`cbed8e9` (#4023)](https://github.com/MetaMask/metamask-mobile/pull/4023/commits/cbed8e99a663962897c2d7615872396dc0a991e2)

File: app/components/UI/TransactionElement/index.js [hunk 7]
```diff
@@ -361,8 +376,8 @@ class TransactionElement extends PureComponent {
 			<>
 				<TouchableHighlight
 					style={styles.row}
-					onPress={this.onPressItem}
 					underlayColor={colors.background.alternative}
+					onPress={() => this.handleTransactionDetailsNavigation('1234')}
```
Comment (wachunei) [inline app/components/UI/TransactionElement/index.js:380]: We should remove every change to this file

File: app/components/Views/FiatOrdersView/index.js [hunk 8]
```diff
@@ -1,4 +1,5 @@
-import React from 'react';
+/* eslint-disable react/prop-types */
```
Comment (wachunei) [inline app/components/Views/FiatOrdersView/index.js:1]: Remove this comment and address the issue, what is it?

File: app/components/UI/FiatOnRampAggregator/Views/TransactionDetails.js (hunk 2 above)
Comment (alaahd) [reply app/components/UI/FiatOnRampAggregator/Views/TransactionDetails.js]: Fixed

File: app/components/Views/FiatOrdersView/index.js (hunk 1 above)
Comment (alaahd) [reply app/components/Views/FiatOrdersView/index.js]: Fixed

File: app/components/UI/FiatOnRampAggregator/components/Quote.tsx (hunk 3 above)
Comment (alaahd) [reply app/components/UI/FiatOnRampAggregator/components/Quote.tsx]: Fixed

File: app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx (hunk 4 above)
Comment (alaahd) [reply app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx:66]: @VikChawla

File: app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx (hunk 5 above)
Comment (alaahd) [reply app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx:190]: @VikChawla

Comment (alaahd) [reply app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx:190]: @VikChawla

File: app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx (hunk 6 above)
Comment (alaahd) [reply app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx:203]: @VikChawla

File: app/components/UI/TransactionElement/index.js (hunk 7 above)
Comment (alaahd) [reply app/components/UI/TransactionElement/index.js:380]: @VikChawla

File: app/components/Views/FiatOrdersView/index.js (hunk 8 above)
Comment (alaahd) [reply app/components/Views/FiatOrdersView/index.js]: Fixed

**Pull request merged**